src/core/ext/transport/chttp2/transport/hpack_encoder.cc

```diff
@@ -380,10 +382,11 @@ static void emit_lithdr_incidx(grpc_chttp2_hpack_compressor* c,
   uint32_t len_val_len;
   GPR_ASSERT(len_val <= UINT32_MAX);
   len_val_len = GRPC_CHTTP2_VARINT_LENGTH((uint32_t)len_val, 1);
-  GRPC_CHTTP2_WRITE_VARINT(key_index, 2, 0x40,
-                           add_tiny_header_data(st, len_pfx), len_pfx);
+  GPR_DEBUG_ASSERT(len_pfx + len_val_len < GRPC_SLICE_INLINED_SIZE);
```

Comment:
was this assumption already made or is it a new assumption? I'm worried that this assumption is making things brittle

Reply:
The initial assumption, though there was no debug assert for it, is that the size of an inlined slice is > than the binary encoding of a single qword + 2 bytes (6 total).

Since the assumption was not explicit, theoretically if inline slice buffers get really tiny, we'd have issues.
Here I'm just making the assumption explicit. Inline slices have 23 bytes max capacity, and here we use 12.

In terms of being brittle: the only way this breaks is if someone reduces the size of inline slice data segments to be smaller than the size of a refcounted slice, which I consider very unlikely.